lib/ip-locator.js

```diff
@@ -0,0 +1,56 @@
+'use strict'
+
+const Ipinfo = require('ipinfo')
+const RequestIp = require('request-ip')
+
+class IpLocator {
+  constructor (options) {
+    const { enabledByDefault = true } = options
+    this.enabledByDefault = enabledByDefault
+  }
+
+  async handle (request, h) {
+    if (this.isDisabledFor(request)) {
+      return h.continue
+    }
+    request.location = await this.locate(request)
+
+    return h.continue
+  }
+
+  routeConfig (request) {
+    return request.route.settings.plugins['hapi-geo-locate'] || {}
+  }
+
+  isDisabledFor (request) {
+    const { enabled = this.enabledByDefault } = this.routeConfig(request)
```

Comment:
This is wrong. You want to check if it is disabled, not enabled.

Please don't blindly pick up my sample code. Create a new method `isEnabledFor(request)` that returns whether geo-locating the request is enabled. You can then use the `isEnabledFor` method within the `isDisabledFor` method

Do you know what I mean?

Reply:
ohk